frontend/public/components/modals/configure-cluster-upstream-modal.tsx

```diff
@@ -0,0 +1,118 @@
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '../factory/modal';
+import { ExternalLink, HandlePromiseProps, withHandlePromise } from '../utils';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { RadioInput } from '../radio';
+import {
+  CLUSTER_VERSION_DEFAULT_UPSTREAM_SERVER_URL,
+  CLUSTER_VERSION_CUSTOM_UPSTREAM_SERVER_URL_PLACEHOLDER,
+} from '@console/shared/src/constants';
+import { TextInput } from '@patternfly/react-core';
+
+export const ConfigureClusterUpstreamModal = withHandlePromise(
+  (props: ConfigureClusterUpstreamModalProps) => {
+    const { cv, handlePromise, close } = props;
+    const currentUpstream = cv?.spec?.upstream;
+
+    const [selectedRadio, setSelectedRadio] = React.useState(
+      currentUpstream ? 'custom' : 'default',
+    );
+    const [customUrl, setCustomUrl] = React.useState(currentUpstream);
+    const [invalidCustomUrl, setInvalidCustomUrl] = React.useState(false);
+
+    const submit: React.FormEventHandler<HTMLFormElement> = (e) => {
+      e.preventDefault();
+      if (selectedRadio === 'custom') {
+        if (!customUrl) {
+          setInvalidCustomUrl(true);
+          return;
+        } else if (customUrl === currentUpstream) {
+          return handlePromise(Promise.resolve(), close);
+        }
+      } else if (!currentUpstream) {
+        return handlePromise(Promise.resolve(), close);
+      }
+      const value = selectedRadio === 'default' ? null : customUrl;
+      const patch = [{ op: 'add', path: '/spec/upstream', value }];
+      return handlePromise(k8sPatch(ClusterVersionModel, cv, patch), close);
+    };
+    const { t } = useTranslation();
+
+    return (
+      <form onSubmit={submit} name="form" className="modal-content ">
+        <ModalTitle>{t('public~Edit upstream configuration')}</ModalTitle>
+        <ModalBody>
+          <p>
+            {t(
+              'public~Select a configuration to receive updates. Updates can be configured to receive information from Red Hat or a custom update service.',
+            )}
+          </p>
+          <ExternalLink
+            href={'/'} //TODO
+            text={t('public~Learn more about OpenShift local update services.')}
+          />
+          <RadioInput
+            onChange={(e) => {
+              setSelectedRadio(e.target.value);
+              setInvalidCustomUrl(false);
+            }}
+            value={'default'}
+            checked={selectedRadio === 'default'}
+            title={t('public~Default. Receive update information from Red Hat.')}
+          >
+            <TextInput
+              id={'cluster-version-default-upstream-server-url'}
+              type="url"
+              isReadOnly
+              isDisabled
+              value={CLUSTER_VERSION_DEFAULT_UPSTREAM_SERVER_URL}
+            />
+          </RadioInput>
+          <RadioInput
+            onChange={(e) => setSelectedRadio(e.target.value)}
+            value={'custom'}
```

Comment:
I'd create constants for `default` and `custom` since they're used in a few places.